Sources/SmokeDynamoDB/AWSDynamoDBCompositePrimaryKeyTable+updateItems.swift

```diff
@@ -139,48 +139,69 @@ public extension AWSDynamoDBCompositePrimaryKeyTable {
-
-        guard let responses = result.responses else {
-            self.logger.warning("BatchExecuteStatementOutput: \(result) does not contain responses")
-            return [:]
+            let promise = self.eventLoop.makePromise(of: [Int: BatchStatementError].self)
+            promise.succeed([:])
+            return promise.futureResult
         }
         
-        var failedList: [Int: BatchStatementError] = [:]
-
-        for (index, response) in responses.enumerated() {
-            if let error = response.error {
-                failedList[index] = error
+        do {
+            let statements: [BatchStatementRequest] = try entries.map { try entryToBatchStatementRequest( $0 ) }
+            let executeInput = BatchExecuteStatementInput(statements: statements)
+            return dynamodb.batchExecuteStatement(input: executeInput).flatMapThrowing { result -> [Int: BatchStatementError] in
+                guard let responses = result.responses else {
+                    self.logger.warning("BatchExecuteStatementOutput: \(result) does not contain responses")
+                    return [:]
+                }
+                var failedList: [Int: BatchStatementError] = [:]
+                for (index, response) in responses.enumerated() {
+                    if let error = response.error {
+                        failedList[index] = error
+                    }
+                }
+                return failedList
             }
+        } catch {
+            let promise = self.eventLoop.makePromise(of: [Int: BatchStatementError].self)
+            promise.fail(error)
+            return promise.futureResult
         }
-
-        return failedList
     }
 
-    func monomorphicBulkWriteWithoutThrowing<AttributesType, ItemType>(_ entries: [WriteEntry<AttributesType, ItemType>]) async throws
-    -> [Int: BatchStatementError] {
+    func monomorphicBulkWriteWithoutThrowing<AttributesType, ItemType>(_ entries: [WriteEntry<AttributesType, ItemType>])
+    -> EventLoopFuture<[Int: BatchStatementError]> {
         // BatchExecuteStatement has a maximum of 25 statements
         // This function handles pagination internally.
         let chunkedEntries = entries.chunked(by: maximumUpdatesPerExecuteStatement)
-        var result: [Int: BatchStatementError] = [:]
-        try await chunkedEntries.enumerated().concurrentForEach { (index, chunk) in
-            try await self.writeChunkedItemsWithoutThrowing(chunk).forEach { (key, val) in
-                result[index * maximumUpdatesPerExecuteStatement + key] = val
-            }
+
+        let futures = chunkedEntries.enumerated().map { (index, chunk) in
+            return self.writeChunkedItemsWithoutThrowing(chunk)
         }  
-        return result
+        
+        do {
+            var failedList: [Int: BatchStatementError] = [:]
+            let results = try EventLoopFuture.whenAllComplete(futures, on: self.eventLoop).wait()
```

Comment:
We shouldn't use `wait()` as this will block the `EventLoop`. You can chain onto this by using `flatMap` instead